core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -657,27 +637,81 @@ public void validateIndexSettings(String indexName, Settings settings) throws In
         return nodesToAllocateOn;
     }
 
-    static void prepareShrinkIndexSettings(ClusterState currentState, Set<String> mappingKeys, Settings.Builder indexSettingsBuilder, Index shrinkFromIndex, String shrinkIntoName) {
-        final IndexMetaData sourceMetaData = currentState.metaData().index(shrinkFromIndex.getName());
+    static void validateSplitIndex(ClusterState state, String sourceIndex,
+                                   Set<String> targetIndexMappingsTypes, String targetIndexName,
+                                   Settings targetIndexSettings) {
+        IndexMetaData sourceMetaData = validateResize(state, sourceIndex, targetIndexMappingsTypes, targetIndexName, targetIndexSettings);
+        IndexMetaData.selectSplitShard(0, sourceMetaData, IndexMetaData.INDEX_NUMBER_OF_SHARDS_SETTING.get(targetIndexSettings));
+        if (sourceMetaData.getCreationVersion().before(Version.V_6_0_0_alpha1)) {
+            // ensure we have a single type since this would make the splitting code considerably more complex
+            // and a 5.x index would not be splittable unless it has been shrunk before so rather opt out of the complexity
+            throw new IllegalStateException("source index created version is too old to apply a split operation");
+        }
+
+    }
+
+    static IndexMetaData validateResize(ClusterState state, String sourceIndex,
+                                           Set<String> targetIndexMappingsTypes, String targetIndexName,
+                                           Settings targetIndexSettings) {
+        if (state.metaData().hasIndex(targetIndexName)) {
+            throw new ResourceAlreadyExistsException(state.metaData().index(targetIndexName).getIndex());
+        }
+        final IndexMetaData sourceMetaData = state.metaData().index(sourceIndex);
+        if (sourceMetaData == null) {
+            throw new IndexNotFoundException(sourceIndex);
+        }
+        // ensure index is read-only
+        if (state.blocks().indexBlocked(ClusterBlockLevel.WRITE, sourceIndex) == false) {
+            throw new IllegalStateException("index " + sourceIndex + " must be read-only to resize index. use \"index.blocks.write=true\"");
+        }
+
+        if ((targetIndexMappingsTypes.size() > 1 ||
+            (targetIndexMappingsTypes.isEmpty() || targetIndexMappingsTypes.contains(MapperService.DEFAULT_MAPPING)) == false)) {
+            throw new IllegalArgumentException("mappings are not allowed when resizing indices" +
+                ", all mappings are copied from the source index");
+        }
+
+        if (IndexMetaData.INDEX_NUMBER_OF_SHARDS_SETTING.exists(targetIndexSettings)) {
```

Comment:
I think that by the time we get here, this setting must be set (it's done in `TransportResizeAction#prepareCreateIndexRequest`)